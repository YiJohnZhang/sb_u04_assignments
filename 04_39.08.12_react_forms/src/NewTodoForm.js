import React, {useState} from 'react';

function NewTodoForm({createToDoEntry}){

	const INTIIAL_FORM_STATE = {
		todoText: ''
	};

	const [formState, setFormState] = useState(INTIIAL_FORM_STATE);
	
	function handleChange(evt){

		const {name, value} = evt.target;
		setFormState({
			[name]: value
		});

	}

	function handleSubmit(evt){

		evt.preventDefault();
		createToDoEntry(formState);
		setFormState(INTIIAL_FORM_STATE);

	}

	return (

		<form>

			<label htmlFor="todoText"><strong>Todo Entry</strong>: </label>
			<input name="todoText"
				type="text"
				data-testid="Todo-input_text"
				placeholder="entry..."
				value={formState.todoText}
				onChange={handleChange}
				/>

			<button data-testid="createTodo" onClick={handleSubmit}>New Item</button>


		</form>

	);

}

export default NewTodoForm;