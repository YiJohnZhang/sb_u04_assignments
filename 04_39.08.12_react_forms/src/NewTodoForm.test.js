import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('smoke test', () => {

	render(<NewTodoForm />);

});

test('snapshot: empty form', () => {
	
	const {asFragment} = render(<NewTodoForm />);
	expect(asFragment()).toMatchSnapshot();

});