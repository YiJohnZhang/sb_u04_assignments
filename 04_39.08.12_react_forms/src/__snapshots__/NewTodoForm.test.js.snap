// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: empty form 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todoText"
    >
      <strong>
        Todo Entry
      </strong>
      : 
    </label>
    <input
      data-testid="Todo-input_text"
      name="todoText"
      placeholder="entry..."
      type="text"
      value=""
    />
    <button
      data-testid="createTodo"
    >
      New Item
    </button>
  </form>
</DocumentFragment>
`;
